### PR TITLE
Command handler to close block comment on /

### LINKED ISSUE #30089: Automatically close block comment on /
I was encouraged by work from @Neme12 in #30074, so I'm resurrecting a *very* old feature request of mine. 😄 

**Steps to Reproduce**:

Press <kbd>/</kbd> in the location marked with `|`:

```csharp
/*
 * Typed some text and then pressed enter...
 * |
```

**Expected Behavior**:

```csharp
/*
 * Typed some text and then pressed enter...
 */
```

**Actual Behavior**:

```csharp
/*
 * Typed some text and then pressed enter...
 * /
```

### PR DESCRIPTION
**Pull request by Neme12**
Fixes #30089

Tagging @sharwell

### CODE REVIEW

**Comment by Neme12** [comment]
retest windows_release_vs-integration_prtest please

File: src/EditorFeatures/CSharp/BlockCommentEditing/CloseBlockCommentCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.VisualStudio.Utilities;
+using VSCommanding = Microsoft.VisualStudio.Commanding;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
```
Comment (DustinCampbell) [inline src/EditorFeatures/CSharp/BlockCommentEditing/CloseBlockCommentCommandHandler.cs]: nit: sort usings

File: src/EditorFeatures/CSharp/BlockCommentEditing/CloseBlockCommentCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        public bool ExecuteCommand(TypeCharCommandArgs args, VSCommanding.CommandExecutionContext executionContext)
+        {
+            if (args.TypedChar == '/')
+            {
+                var caret = args.TextView.GetCaretPoint(args.SubjectBuffer);
+                if (caret != null)
+                {
+                    var (snapshot, position) = caret.Value;
+
+                    if (position >= 2 &&
```
Comment (DustinCampbell) [inline src/EditorFeatures/CSharp/BlockCommentEditing/CloseBlockCommentCommandHandler.cs]: Suggestion: this logic, and the nested logic that checks to see if the line before the '*' is whitespace could use a comment to describe the behavior.

File: src/EditorFeatures/Text/Shared/Extensions/ITextSnapshotLineExtensions.cs [hunk 3]
```diff
@@ -67,13 +67,18 @@ internal static class ITextSnapshotLineExtensions
         /// <summary>
         /// Determines whether the specified line is empty or contains whitespace only.
         /// </summary>
-        public static bool IsEmptyOrWhitespace(this ITextSnapshotLine line)
+        public static bool IsEmptyOrWhitespace(this ITextSnapshotLine line, int startIndex = 0, int endIndex = -1)
```
Comment (DustinCampbell) [inline src/EditorFeatures/Text/Shared/Extensions/ITextSnapshotLineExtensions.cs]: Maybe add an argument check to validate that `startIndex` is not less than zero?

**Comment by Neme12** [comment]
@DustinCampbell 


> 1\. Does undo still work as expected?

I'm not sure what your expected behavior is. After undo, the `/` is removed, and the state of the world is as before pressing the key. There is no intermediate step when undoing (that would put the `/` in the position you actually typed it in).

> 2\. What if there is text like so, but `Hello` is selected from right to left (so that the caret is on the left)?

This command handler isn't activated (because there is a selection).

> 3\. What happens if there's multiple selections in the editor?

This command will only work on the primary cursor (colored in red).

I tried to use the new multi-caret API, but found it rather confusing and harder to use.
1. There is no simple API to get the caret positions (unlike for a single caret) - we'd have to go through the selection and check their sizes.
2. I found 3 different properties for getting the selections and didn't have an idea as to what the difference is and which one I should use. 2 of those 3 properties had a warning about being costly or computationally expensive.

File: src/EditorFeatures/CSharp/BlockCommentEditing/CloseBlockCommentCommandHandler.cs (hunk 1 above)
Comment (Neme12) [reply src/EditorFeatures/CSharp/BlockCommentEditing/CloseBlockCommentCommandHandler.cs]: ➡️ Fixed in 42baa98e7fb23c6820fa7dccb247b0fedb7c6374

File: src/EditorFeatures/CSharp/BlockCommentEditing/CloseBlockCommentCommandHandler.cs (hunk 2 above)
Comment (Neme12) [reply src/EditorFeatures/CSharp/BlockCommentEditing/CloseBlockCommentCommandHandler.cs]: ➡️ Comment added in 42baa98e7fb23c6820fa7dccb247b0fedb7c6374

File: src/EditorFeatures/Text/Shared/Extensions/ITextSnapshotLineExtensions.cs (hunk 3 above)
Comment (Neme12) [reply src/EditorFeatures/Text/Shared/Extensions/ITextSnapshotLineExtensions.cs]: ➡️ Check added in 42baa98e7fb23c6820fa7dccb247b0fedb7c6374

**Pull request merged**